core/templates/pages/exploration-editor-page/editor-tab/templates/modal-templates/teach-oppia-modal.controller.ts

```diff
@@ -98,12 +98,12 @@ angular.module('oppia').controller('TeachOppiaModalController', [
           state_name: stateName
         }
       }).then(function(response) {
-        $scope.showUnresolvedAnswers(response.unresolved_answers);
+        $scope.showUnresolvedAnswers(response.data.unresolved_answers);
```

Comment:
Hm, did this not work before?

Reply:
I think it was working indeed but it was not working on unit tests using `$httpBackend`. I've checked and it's working as expected after changes.

Reply:
Couple of questions:
1. Did you dig into why the previous code did not work in unit tests?
2. How is this done in other unit tests?
3. If this is the pattern to be followed for all `response` types, should we audit and change this in other places as well?

Reply:
1. The code wasn't working on unit tests because when mocking a return from `$httpBackend`, the return is passed through the request body, which is accessed with `response.data`. It was failing until fixing `response.unresolved_answers` to `response.data.unresolved_answers`. For example
```js
$httpBackend.expect(...).respond({ foo: 'baz' });

response.foo // undefined
response.data.foo // baz
```
[Here's](https://docs.angularjs.org/api/ng/service/$http#$http-returns) the link for more information.

2. This is achieved in other unit tests (AngularJS ones which use $httpBackend) by just following the same example above (first line). This might confuse you because there are more than one way to mock a response on unit tests:
- By mocking a class method, then we don't need to use data. For example:
```js
spyOn(Class, 'method).and.returnValue({ foo: 'baz' });
```
- It depends on which Angular version you're using (this explanation is only for AngularJS files).
3. This pattern should be followed by ALL AngularJS files and unit tests. For Angular2+ files, it might change a little. I think @bansalnitish and @srijanreddy98 can help with this question. I think creating a lint test can be a bit complex but I think it should a pattern to be followed, we can explain it on the wiki and enforce it through code review.

Reply:
I think the pattern is pretty simple: For angular HTTPClient use `response`, for angularjs httpBackend use `response.data`

Reply:
Souns good, thanks for a clear explanation @marianazangrossi! :)